src/integrated/soil_canopy_model.jl

```diff
@@ -160,7 +174,22 @@ function SoilCanopyModel{FT}(;
         canopy_model_args...,
     )
 
-    return SoilCanopyModel{FT, typeof(soil), typeof(canopy)}(soil, canopy)
+    soilco2 = Soil.Biogeochemistry.SoilCO2Model{FT}(; soilco2_args...)
+
+    # not sure if the if below is needed
+    if typeof(soilco2_args.drivers.met) != PrognosticMet
+        throw(
+            AssertionError(
+                "To run with a prognostic soil energy and hydrology model, the met driver must be of type PrognosticSoil.",
+            ),
+        )
+    end
```

Comment:
can also check that soil porosity is the same between soil & soil co2


if we have other "requirements" we want to enforce, here is the place to do them. E.g. setting the BC for the soil co2 to depend on air co2.